YAFC/Windows/WelcomeScreen.cs

```diff
@@ -61,17 +62,35 @@ private enum EditType {
             Workspace, Factorio, Mods
         }
 
-        public WelcomeScreen() : base(ImGuiUtils.DefaultScreenPadding) {
+        public WelcomeScreen(string[] args = null) : base(ImGuiUtils.DefaultScreenPadding) {
+            if (tips == null) {
+                tips = File.ReadAllLines("Data/Tips.txt");
+            }
+
+            IconCollection.ClearCustomIcons();
             RenderingUtils.SetColorScheme(Preferences.Instance.darkMode);
-            var lastProject = Preferences.Instance.recentProjects.FirstOrDefault();
+
+            if (args != null && args.Length >= 3) {
```

Comment:
That's a lot of logic happening in a ctor. I'd prefer to parse the args outside and pass a `RecentProject` into the `WelcomeScreen` ctor.

On the other hand I don't really know how the other parts of the code are doing it, this is a more general advice.

Reply:
Yes, let the `Main()` handle parsing the CLI arguments and pass a file that should be loaded and/or other configurations to the constructor. (Much better ~for testing (when we are adding this)~ in general)

Reply:
I moved the parameter parsing outside of WelcomeScreen